### PR TITLE
Adding get_flux function to SupernovaModel

### LINKED ISSUE #213: Add "calculate neutrino flux at given distance" to SupernovaModel
## What I want

As a user I would like to get the supernova neutrino flux, predicted by a certain `model` (subclass of `SupernovaModel`) at a given `distance` at given times and energies (in `neutrinos/cm^2/MeV/s`).

## What we have now
Currently `SupernovaModel` class allows only to get neutrino flux in 4pi (`get_initial_spectra`, `get_transformed_spectra`), and has no means for calculating flux at a given distance.

We have these calculations only in `generate_fluence` and `generate_time_series`:
https://github.com/SNEWS2/snewpy/blob/b35379952fe18aedc28cdb106f1ee9e7ad08b082/python/snewpy/snowglobes.py#L122
https://github.com/SNEWS2/snewpy/blob/b35379952fe18aedc28cdb106f1ee9e7ad08b082/python/snewpy/snowglobes.py#L288

which will finally save the flux/fluence to the disk, which is not always needed.

While these calculations are not complicated, we shouldn't make user I think they belong to the SupernovaModel class.

## What I suggest

1. Create a method in the base class `SupernovaModel.get_initial_flux (t, E, distance, **kwargs)`
which by default would call `self.get_initial_spectra`. If there are any additional input parameters for a specific model (direction? flavors?), this method should be reimplemented in the models's class, making use of `**kwargs` argument.

2. This method should return an object, containing the flux for each of the `(Flavor,T,E)` points, and optionally pack

    * Energy points `E`
    * Time points `T`
    * Available flavors 

   In a simplest case it could be `Dict[Flavor, np.array]`, but I propose to discuss this separately in #214

### PR DESCRIPTION
**Pull request by Sheshuk**
Closes #213

### CODE REVIEW

File: python/snewpy/models/base.py [hunk 1]
```diff
@@ -161,6 +162,33 @@ def get_transformed_spectra(self, t, E, flavor_xform):
 
         return transformed_spectra   
 
+    def get_flux (self, t, E, distance, flavor_xform=NoTransformation(), **kwargs):
```
Comment (JostMigenda) [inline python/snewpy/models/base.py:165]:
```suggestion
    def get_flux (self, t, E, distance, flavor_xform=NoTransformation()):
```
If the `**kwargs` aren’t used (or even passed through to some other function), I would delete them here.

File: python/snewpy/models/base.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
 from scipy.special import loggamma
 
 from snewpy.neutrino import Flavor
+from snewpy.flavor_transformation import NoTransformation, FlavorTransformation
```
Comment (JostMigenda) [inline python/snewpy/models/base.py:12]:
```suggestion
from snewpy.flavor_transformation import NoTransformation
```
I don’t think `FlavorTransformation` is used anywhere in this PR?

File: python/snewpy/models/base.py [hunk 3]
```diff
@@ -161,6 +162,33 @@ def get_transformed_spectra(self, t, E, flavor_xform):
...
+        """Get neutrino flux through 1cm^2 surface at the given distance
+
+        Parameters
+        ----------
+        t : astropy.Quantity
+            Time to evaluate the neutrino spectra.
+        E : astropy.Quantity or ndarray of astropy.Quantity
+            Energies to evaluate the the neutrino spectra.
+        distance : astropy.Quantity or float (in kpc)
+            Distance from supernova to the
```
Comment (JostMigenda) [inline python/snewpy/models/base.py:175]:
```suggestion
            Distance from supernova.
```

File: python/snewpy/models/base.py [hunk 4]
```diff
@@ -161,6 +162,33 @@ def get_transformed_spectra(self, t, E, flavor_xform):
...
+            Energies to evaluate the the neutrino spectra.
+        distance : astropy.Quantity or float (in kpc)
+            Distance from supernova to the 
+        flavor_xform : FlavorTransformation
+            An instance from the flavor_transformation module.
+
+        Returns
+        -------
+        dict
+            Dictionary of neutrino fluxes in [neutrinos/(cm^2*MeV*s)],
```
Comment (JostMigenda) [inline python/snewpy/models/base.py:182]:
```suggestion
            Dictionary of neutrino fluxes in [neutrinos/(cm^2*erg*s)], 
```
`get_transformed_spectra` returns `<Quantity 1.234e+56 1 / (erg s)>`

**Review by JostMigenda — COMMENTED** [review]
Looks good to me, just some very minor comments

File: python/snewpy/models/base.py (hunk 1 above)
Comment (Sheshuk) [reply python/snewpy/models/base.py:165]: You're right. What I had in mind is to be compatible with all the models, including Fornax etc. But this is a separate issue #223 
For this PR I should remove this.

**Review by JostMigenda — APPROVED** [review]

**Pull request merged**